src/lib.rs

```diff
@@ -45,6 +45,9 @@ pub fn run_program_with_custom_state(bin_path: &str, mut vm: VMState) -> (U256,
         let opcode = vm.get_opcode(&opcode_table);
         if vm.predicate_holds(&opcode.predicate) {
             match opcode.variant {
+                // TODO: Properly handle what happens
+                // when the VM runs out of ergs/gas.
+                _ if vm.gas_left() == 0 => break,
```

Comment:
Why inside the match?

Reply:
I thought it was the best place to check on how much gas the VM had left before actually executing an instruction, do you think otherwise?
Any suggestions on where to place this check?

Reply:
I would put it just after the loop, since it doesn't actually use the opcode itself, but its not really that important, this works too